### PR TITLE
Show alerts when errors encountered for suspend/resume

### PR DESCRIPTION
**Pull request by ps2**

### CODE REVIEW

File: DashKit/DashPumpManagerError.swift [hunk 1]
```diff
@@ -12,28 +12,27 @@ import PodSDK
...
         case .invalidBasalSchedule:
-            return LocalizedString("Invalid Basal Schedule.", comment: "Description of registration error for connection timeout.")
-        case .podCommError:
-            return LocalizedString("There was a problem communicating with the pod.", comment: "Description of communication error with pod.")
+            return LocalizedString("Invalid Basal Schedule.", comment: "Description of DashPumpManagerError for .invalidBasalSchedule")
+        case .podCommError(let error):
+            return error.errorDescription
         case .busy:
-            return LocalizedString("Pod Busy.", comment: "Description of registration error when error is unknown.")
+            return LocalizedString("Pod Busy.", comment: "Description of DashPumpManagerError error when pump manager is busy.")
```
Comment (nhamming) [inline DashKit/DashPumpManagerError.swift:28]: [nit] the comment breaks the pattern established in the other cases.

**Review by nhamming — APPROVED** [review]
LGTM

File: DashKit/DashPumpManager.swift [hunk 2]
```diff
@@ -887,9 +887,13 @@ public class DashPumpManager: PumpManager {
...
         }
     }
-    
+
     public func resumeDelivery(completion: @escaping (Error?) -> Void) {
-        let preflightError = self.setStateWithResult({ (state) -> Error? in
+        self.resumeInsulinDelivery(completion: completion)
+    }
+
+    public func resumeInsulinDelivery(completion: @escaping (DashPumpManagerError?) -> Void) {
```
Comment (darinkrauss) [inline DashKit/DashPumpManager.swift:895]: [nit] "resumeInsulinDelivery" inconsistent with "suspendDelivery".

File: DashKitUI/Protocols/DashSettingsViewModelProtocol.swift [hunk 3]
```diff
@@ -16,7 +16,17 @@ struct BasalDeliveryRate {
...
 }
 
+enum DashSettingsViewAlert {
+    case suspendError(DashPumpManagerError)
+    case resumeError(DashPumpManagerError)
+}
+
 protocol DashSettingsViewModelProtocol: ObservableObject, Identifiable {
+    
+    var activeAlert: DashSettingsViewAlert? { get }
```
Comment (darinkrauss) [inline DashKitUI/Protocols/DashSettingsViewModelProtocol.swift:26]: [nit] Does this need to be settable?

**Review by darinkrauss — APPROVED** [review]
A couple of nits, but otherwise LGTM! 👾

File: DashKitUI/Protocols/DashSettingsViewModelProtocol.swift (hunk 3 above)
Comment (ps2) [reply DashKitUI/Protocols/DashSettingsViewModelProtocol.swift:26]: No, not to fulfill the protocol.

**Pull request merged**